components/script_traits/lib.rs

```diff
@@ -460,6 +460,19 @@ pub enum MouseEventType {
     MouseUp,
 }
 
+/// The Wheel event deltas in every direction
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
+pub struct WheelDelta {
+    /// Delta in the left/right direction
+    pub x: f64,
+    /// Delta in the up/down direction
+    pub y: f64,
+    /// Delta in the direction going into/out of the screen
+    pub z: f64,
+    /// Mode to measure the floats in
+    pub mode: u32,
```

Comment:
Maybe use an enum + FromPrimitive.

Reply:
Not sure exactly what you mean. How would I implement FromPrimitive for this vector? Also are you thinking something like this?
```
enum WheelDelta {
    ModeDeltaPixel { x: f64, y: f64, z: f64 },
    ModeDeltaLine { x: f64, y: f64, z: f64 },
    ModeDeltaPage { x: f64, y: f64, z: f64 },
}
```

Reply:
I meant for the `mode` field.

Reply:
Changed to `pub mode: WheelMode` and `enum WheelMode { Pixel = 0x00 }`. Didn't do a FromPrimitive impl